src/main/java/com/javadiscord/javabot/StatsUpdater.java

```diff
@@ -0,0 +1,44 @@
+package com.javadiscord.javabot;
+
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.OnlineStatus;
+import net.dv8tion.jda.api.events.ReadyEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class StatsUpdater extends ListenerAdapter {
+
+    private final ScheduledExecutorService threadPool = Executors.newSingleThreadScheduledExecutor();
```

Comment:
You should use `Bot.asyncPool` instead of creating another thread pool here. That thread pool already has I think 4 threads so it should be more than enough.